src/Admin/SiteHealth.php

```diff
@@ -76,11 +66,20 @@ public function __construct( MonitorCssTransientCaching $css_transient_caching,
 	 * Adds the filters.
 	 */
 	public function register() {
-		add_filter( 'site_status_tests', [ $this, 'add_tests' ] );
-		add_filter( 'debug_information', [ $this, 'add_debug_information' ] );
-		add_filter( 'site_status_test_result', [ $this, 'modify_test_result' ] );
-		add_filter( 'site_status_test_php_modules', [ $this, 'add_extensions' ] );
-		add_action( 'admin_print_styles-site-health.php', [ $this, 'add_styles' ] );
+
+		if ( is_admin() ) {
+			add_filter( 'site_status_tests', [ $this, 'add_tests' ] );
+			add_filter( 'debug_information', [ $this, 'add_debug_information' ] );
+			add_filter( 'site_status_test_result', [ $this, 'modify_test_result' ] );
+			add_filter( 'site_status_test_php_modules', [ $this, 'add_extensions' ] );
+			add_action( 'admin_print_styles-site-health.php', [ $this, 'add_styles' ] );
+		}
+
+		if ( wp_doing_ajax() ) {
+			add_action( 'wp_ajax_health-check-site-status', [ $this, 'ajax_site_status' ], 11 );
+		}
+
+		$this->maybe_send_random_number();
```

Comment:
Maybe only run if not a AJAX request?

```suggestion
		if ( ! wp_doing_ajax() ) {
			$this->maybe_send_random_number();
		}
```